glommio/src/sys/uring.rs

```diff
@@ -352,7 +352,7 @@ where
                 let mut buf = buffer_allocation(len).expect("Buffer allocation failed");
                 let src = peek_source(from_user_data(op.user_data));
 
-                match &mut *src.source_type.borrow_mut() {
+                match &mut src.borrow_mut().source_type {
```

Comment:
Note that if this is broken and clippy doesn't complain because of the `unsafe` block, it would be very hard to catch. WE had issues before with things dying in the io_uring loop and because most of the time the requests complete right away, we don't notice. Until they don't...

Reply:
Can you explain why this isn't safe? The borrow is src and it lives long enough. The compiler allows the borrow to be in the `match` statement because it can guarantee none of the inner data escapes the scope.

I also tried to remove the unsafe bit and replaced it by many minimal ones  and the compiler doesn't take offense there either.

Reply:
I have rebased my branch and I no longer use this pattern. I now give the borrow its own, separate lifetime.